src/backends/backend.js

```diff
@@ -70,8 +71,12 @@ class Backend {
   currentUser() {
     if (this.user) { return this.user; }
     const stored = this.authStore && this.authStore.retrieve();
-    if (stored) {
-      return Promise.resolve(this.implementation.setUser(stored)).then(() => stored);
+    if (stored && (stored.backend === this.backendName)) {
```

Comment:
No parens necessary on the equality check. Also, keep the naming synced - if it's `this.backendName`, let's call it `backendName` in the stored user.

Reply:
Done.